src/objectExplorer/objectExplorerService.ts

```diff
@@ -191,7 +195,13 @@ export class ObjectExplorerService {
         }
     }
 
-    public updateNode(node: TreeNodeInfo): void {
+    public updateNode(node: TreeNodeInfo | ConnectTreeNode): void {
+        if (node instanceof ConnectTreeNode) {
+            node = node.parentNode;
+            if (!(node instanceof TreeNodeInfo)) {
```

Comment:
1. Is there a reason you added this check here but not anywhere else? Do you actually expect this to be able to happen ever? I don't have a huge issue with being careful, but maybe this would be better put in a helper function (`getParentTreeNode` or something)
2. If this error ends up being displayed to the user it should be localized

Reply:
yeah, I was mostly being careful

Reply:
The error is thrown here, but what happens in the UI? Would the user know there was an error in the one of the nodes?

Reply:
Do you think I should have a popup to the user in addition to the error message? @abist

Reply:
Either that, or put that error in the node

Reply:
Having a popup seems to be the cleanest option, I think trying to fit text in the node would be hard to read and not immediately indicate an issue occurred.